### ISSUE TITLE
Bad autocorrect for Style/CaseLikeIf

### ISSUE THREAD

**Issue by jcoyne** [rubocop-hq/rubocop#8508]
In rubocop 0.89.0

If you have code like this:

```ruby
if klass == ClassA
  do_something
elsif klass == ClassB
  do_something_else
end
```

It incorrectly gets autocorrected to:

```ruby
 case klass
 when ClassA
   do_something
 when ClassB
   do_something_else
 end
```

@marcandre warned about this here https://github.com/rubocop-hq/rubocop/issues/7736#issuecomment-634915216 but I think this advice went unheeded.

**Comment by marcandre** [comment]
This cop wasn't marked as `Safe` either 🤷‍♂️

@fatkodima would you like to handle this?

**Comment by fatkodima** [comment]
Can't reproduce.

```
$ cat test.rb
if klass == ClassA
  do_something
elsif klass == ClassB
  do_something_else
end

$ cat .rubocop.yml
...
Style/CaseLikeIf:
  Enabled: true
....

$ rubocop test.rb -a
Inspecting 1 file
C

Offenses:

test.rb:1:1: C: Style/FrozenStringLiteralComment: Missing frozen string literal comment.
if klass == ClassA
^
test.rb:3:1: C: Style/MissingElse: if condition requires an else-clause.
elsif klass == ClassB ...
^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 2 offenses detected
```

**Comment by jcoyne** [comment]
Here is the "real" code where autocorrect went wrong for me: https://github.com/sul-dlss/dor-services-app/commit/1a289b15ee128392a7ebeacbb76e0ec69c14c5d3#diff-c03d1fa455324b14d721e59ca497ab9eL27-L35

**Comment by jcoyne** [comment]
So it appears that it might have something to do with when it things things are a class or a constant.

For example, this will trigger it:
```
if klass == Nokogiri::XML
  do_something
elsif klass == ClassB
  do_something_else
end
```
```
test.rb:3:1: C: Style/CaseLikeIf: Convert if-elsif to case-when.
if klass == Nokogiri::XML ...
^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by fatkodima** [comment]
Thanks for the report. Should be fixed by https://github.com/rubocop-hq/rubocop/pull/8511
